### PR TITLE
Hyperlinkify names and aliases in listing tables

### PR DESCRIPTION
**Pull request by callunity**
Instead of having to move the mouse to the far side of the screen to click a tiny pencil icon, users can now click on the Name or Alias of their chosen item to navigate to its page.

### CODE REVIEW

File: miso-web/src/main/webapp/pages/editIlluminaPool.jsp [hunk 1]
```diff
@@ -182,9 +182,11 @@
               <td id="result${qc.id}">${qc.results} ${qc.qcType.units}</td>
               <c:if test="${(library.securityProfile.owner.loginName eq SPRING_SECURITY_CONTEXT.authentication.principal.username)
                                         or fn:contains(SPRING_SECURITY_CONTEXT.authentication.principal.authorities,'ROLE_ADMIN')}">
-                <td id="edit${qc.id}" align="center"><a href="javascript:void(0);"
-                                                        onclick="Pool.qc.changePoolQCRow('${qc.id}','${pool.id}')">
-                  <span class="ui-icon ui-icon-pencil"></span></a></td>
+
+              <%-- GLT-201: Comment to remove 'Edit Column' --%>
```
Comment (apmasell) [inline miso-web/src/main/webapp/pages/editIlluminaPool.jsp]: Please delete this and the other similar comments.

File: miso-web/src/main/webapp/scripts/sample_ajax.js [hunk 2]
```diff
@@ -436,7 +436,7 @@ Sample.library = {
     }
   },
   
-  validateLibraryQcs: function () {
+  validateLibraryQcs: function (json) {
```
Comment (apmasell) [inline miso-web/src/main/webapp/scripts/sample_ajax.js]: Why did this change?

Comment (callunity) [inline miso-web/src/main/webapp/scripts/sample_ajax.js]: It's a fix for a  previous commit (https://github.com/oicr-gsi/miso-lims/commit/494ae1b6ba551fb651be0ee4b497ec6801d2c93a#diff-dc20b4f617c4dd4cdee5a309a1f97d0aL184) in which I refactored the method and accidentally removed the JSON argument.

File: miso-web/src/main/webapp/pages/editIlluminaPool.jsp (hunk 1 above)
Comment (callunity) [inline miso-web/src/main/webapp/pages/editIlluminaPool.jsp]: It will be eliminated.

**Comment by apmasell** [comment]
:+1:

**Comment by callunity** [comment]
Merge conflicts fixed.

File: spring/src/main/java/uk/ac/bbsrc/tgac/miso/spring/ajax/ProjectControllerHelperService.java [hunk 3]
```diff
@@ -365,10 +364,15 @@ public JSONObject listProjectsDataTable(HttpSession session, JSONObject json) {
       final JSONObject j = new JSONObject();
       final JSONArray jsonArray = new JSONArray();
       for (Project project : requestManager.listAllProjects()) {
-        jsonArray.add(JsonSanitizer.sanitize("[\"" + project.getName() + "\",\"" + project.getAlias() + "\",\"" + project.getDescription()
-            + "\",\"" + project.getProgress().getKey() + "\",\"" + project.getProjectId() + "\",'" + "<a href=\"/miso/project/"
-            + project.getId() + "\"><span class=\"ui-icon ui-icon-pencil\"></span></a>" + "']"));
+        JSONArray inner = new JSONArray();
 
+        inner.add(TableHelper.hyperLinkify("/miso/project" + project.getId(), project.getName()));
+        inner.add(TableHelper.hyperLinkify("/miso/project" + project.getId(), project.getAlias()));
```
Comment (djcooke) [inline spring/src/main/java/uk/ac/bbsrc/tgac/miso/spring/ajax/ProjectControllerHelperService.java]: Missing slash - should be "/miso/project/" on both lines

File: spring/src/main/java/uk/ac/bbsrc/tgac/miso/spring/ajax/TableHelper.java [hunk 4]
```diff
@@ -0,0 +1,20 @@
+package uk.ac.bbsrc.tgac.miso.spring.ajax;
+
+public class TableHelper {
+
+  // hyperLinkify(String, String) returns the hyperlink-ed (in HTML) version of
+  // text given a path.
```
Comment (djcooke) [inline spring/src/main/java/uk/ac/bbsrc/tgac/miso/spring/ajax/TableHelper.java]: Is this supposed to be Javadoc?

Comment (callunity) [inline spring/src/main/java/uk/ac/bbsrc/tgac/miso/spring/ajax/TableHelper.java]: Good question. Likely, but it looks like it was written with JS styling. I'll fix that.

**Comment by djcooke** [comment]
:+1:

**Pull request merged**